File: ruby/lib/agentic_commerce_protocol/models/allowance_reason.rb
# frozen_string_literal: true
# This file is @generated
module AgenticCommerceProtocol
  class AllowanceReason
    ONE_TIME = "one_time".freeze

    def self.all_vars
      @all_vars ||= [ONE_TIME].freeze
    end

    def initialize(value)
      unless AllowanceReason.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #AllowanceReason"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if AllowanceReason.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #AllowanceReason"
    end

    def serialize
      @value
    end
  end
end

File: rust/src/models/allowance_reason.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum AllowanceReason {
    #[serde(rename = "one_time")]
    OneTime,
}

impl fmt::Display for AllowanceReason {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::OneTime => "one_time",
        };
        f.write_str(value)
    }
}
